e2e/node_config.go

```diff
@@ -156,8 +155,8 @@ func baseTemplateLinux(location string) *datamodel.NodeBootstrappingConfiguratio
 						UserAssignedID:                    "",
 						UserAssignedClientID:              "",
 						CustomHyperkubeImage:              "",
-						CustomKubeProxyImage:              "mcr.microsoft.com/oss/kubernetes/kube-proxy:v1.26.0.1",
-						CustomKubeBinaryURL:               "https://acs-mirror.azureedge.net/kubernetes/v1.26.0/binaries/kubernetes-node-linux-amd64.tar.gz",
+						CustomKubeProxyImage:              "mcr.microsoft.com/oss/kubernetes/kube-proxy:v1.27.16", // Oldest supported version, end of life is Jul 2025 https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli#kubernetes-versions
```

Comment:
why do we need to use the latest supported version ?

Reply:
I think we should test something we support and actually run in production, rather than something has been deprecated a while ago.

Reply:
i think it would worthwhile to resolve this version dynamically at runtime by inspecting the e2e cluster's properties and extracting the exacting the control plane's version

Reply:
added

Reply:
I meant to write the oldest :). 

I was looking to have Artur come up with Cameron suggestion   Wednesday are all set